magento/magento2
# Module data fixtures for @magentoDataFixtureBeforeTransaction annotations

### Description (*)

Extend the option to specify a module based data fixture files for integration tests to the `@magentoDataFixtureBeforeTransaction` annotation.
Previously they where only available for the `@magentoDataFixture` annotation.

The syntax is the same:
```
@magentoDataFixtureBeforeAnnotation Foo_DataFixtureDummy::Test/Integration/foo.php
```

There is no backward compatibility break associated with this commit.

The PR also adds previously missing test coverage for same functionality for the `@magentoDataFixture` test annotation. Since the code from this PR is identical to that code, I decided against adding a test for `@magentoDataFixtureBeforeTransaction`, too.

### Fixed Issues (if relevant)

There is no relevant open issue I'm aware of.

### Manual testing scenarios (*)

1. Create module
2. Create a Integration Test within the module
3. Add a annotation pointing to a file within the module, for example:
   `@magentoDataFixtureBeforeTransaction Magento_InventoryCatalog::Test/_file/products_all_types.php`
4. Run the test and observe the fixture being applied.

### Contribution checklist (*)
